custom/icds_reports/models/aggregate.py

```diff
@@ -1470,3 +1472,26 @@ class Meta(object):
 
     _agg_helper_cls = AwwIncentiveAggregationDistributedHelper
     _agg_atomic = False
+
+
+class DashboardUserActivityReport(models.Model, AggregateMixin):
+    """
+    Daily Update table to hold Dashboard users activity information
+    """
+    username = models.TextField(primary_key=True)
```

Comment:
I think that this usage of primary key is quite confusing. Generally I would expect `Report.objects.get(username='user')` to return one result if this were a "true" primary key. However since this report is using child tables, that username could return one record for every date. This is because primary keys are checked between child tables (I think that's 'fixed' if you use pg12's declarative partitioning, but that's not applicable here). Is that intentional? is there a particular reason to keep the old data around, or could we just keep the most recent day?

If we need to keep the historical data, then my personal preference would be to have a unique index on this with a comment about how this is only unique for the day. If we don't need to keep that historical data, then I think that it would be nice to not use table inheritance in this case

Reply:
Yeah, the decision was intentional to be more future proof with coming feature change-requests which I discussed with GS and data pull requests around this which GS told we should expect pretty frequently in future.

Reply:
On keeping the historical data. you think I should keep index on user_name and date or just on username or just on date. Asking in terms of query performance if there is a query like  `select * from table where date=XYZ` or `select * from table where username in (...) and date=XYZ`

Reply:
I see now. Checked the other aggregation tables(agg_awc_daily which resembles it closely) and found that indexes is only created on columns(other than the date) and on the column date separately not in together.  Also, the reason for creating a column on date as well in agg_awc_daily would be most probably because its a huge child table created every day.

Also, this primary key on the username will create a index on it and the number of rows in this particular table is significantly lower so I think it should not be needed to create another index on the date as well. So leaving the index limited to username only(As you said)

Reply:
Confirmed on Skype. Only valid user with multiple locations will be CPMU user which is not a part of this report. So in this report users will have only one location assigned

Reply:
> On keeping the historical data. you think I should keep index on user_name and date or just on username or just on date. Asking in terms of query performance if there is a query like select * from table where date=XYZ or select * from table where username in (...) and date=XYZ

To be clear, the reason that you should not index date is because date is included in the child table's constraint. That means that tables that are not the date for that table are filtered out by postgres's constraint exclusion logic and do not need an index to efficiently filter them https://www.postgresql.org/docs/11/ddl-partitioning.html#DDL-PARTITIONING-CONSTRAINT-EXCLUSION